pkg/kv/kvserver/store_pool.go

```diff
@@ -152,37 +152,36 @@ func MakeStorePoolNodeLivenessFunc(nodeLiveness *liveness.NodeLiveness) NodeLive
 //
 //  - Let's say a node write its liveness record at tWrite. It sets the
 //    Expiration field of the record as tExp=tWrite+livenessThreshold.
-//    The node is considered LIVE (or DECOMMISSIONING or DRAINING).
+//    The node is considered LIVE.
 //  - At tExp, the IsLive() method starts returning false. The state becomes
-//    UNAVAILABLE (or stays DECOMMISSIONING or DRAINING).
-//  - Once threshold passes, the node is considered DEAD (or DECOMMISSIONED).
+//    UNAVAILABLE.
+//  - Once threshold passes, the node is considered DEAD.
 //
-// NB: There's a bit of discrepancy between what "Decommissioned" represents, as
-// seen by NodeStatusLiveness, and what "Decommissioned" represents as
-// understood by MembershipStatus. Currently it's possible for a live node, that
-// was marked as fully decommissioned, to have a NodeLivenessStatus of
-// "Decommissioning". This was kept this way for backwards compatibility, and
-// ideally we should remove usage of NodeLivenessStatus altogether. See #50707
-// for more details.
+// The membership status of the node is reported separately and is based on the
+// MembershipStatus proto.
 func LivenessStatus(
 	l livenesspb.Liveness, now time.Time, deadThreshold time.Duration,
-) livenesspb.NodeLivenessStatus {
+) (NodeStatus, NodeMembershipStatus) {
```

Comment:
Instead of creating these `NodeMembershipStatus` and `NodeStatus` types, I was hoping we would simply start using the liveness record itself. It has all the information we need.

From #50707.
> (c) kvserver.Allocator: Talking through storeDetail.status, and kvserver.LivenessStatus, it uses both "membership" status and "process" state to inform placement of replicas.
> ...
> For (c), like suggested above, we can simply fetch the liveness record of interest (which captures the expiration time of said liveness record), and use the current time to determine whether or not a given store is a viable placement target for replicas. It can be well insulated between storeDetail.status.

Instead of `NodeMembershipStatus{Decommissioning,Decommissioned,Active}`, we could just use `livenesspb.Liveness.Membership`. To check whether a given store is draining (`NodeMembershipStatusDraining`), we could just check `livenesspb.Liveness.Draining`. I think the NodeMembershipStatus enum is a bit confused. It claims to hold a "membership status", but it's possible for a node to be both active/decommissioning/decommissioned _and_ draining -- state that's well represented already using the liveness proto. I think we should centralize around just the one liveness type from now on.

As for` NodeStatus{Dead,Unavailable,Unknown}`, we could again use the expiration timestamp on the liveness record directly. We're doing it right here to generate these statuses in the first place.

---

All together I think it would be simpler to just use the liveness proto at the few call sites instead of introducing these two types.